components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/subscription/registry/RegistrySubscriptionManager.java

```diff
@@ -196,11 +185,45 @@ private String getResourcePath(String subscriptionID, String topicName) {
     }
 
     /**
-     * Get all the subscriptions. it get the subscription ids from the topic index and get
-     * relavent subscriptions.
-     * @return
-     * @throws org.wso2.carbon.event.core.exception.EventBrokerException
+     * Calculates the JMS subscription stored path for a WSSubscription using subscription id and the topic name
+     *
+     * @param subscriptionID the subscription ID
+     * @param topicName      the topic name
+     * @return the JMS subscription resource path for a subscription
+     */
+    private String getJMSSubResourcePath(String subscriptionID, String topicName) {
+        String resourcePath = this.topicStoragePath;
+
+        // first convert the . to /
+        topicName = topicName.replaceAll("\\.", "/");
+
+        if (!topicName.startsWith("/")) {
+            resourcePath = resourcePath + "/";
+        }
+
+        // this topic name can have # and * marks if the user wants to subscribes to the
+        // child topics as well. but we consider the topic here as the topic name just before any
+        // special character.
+        // eg. if topic name is myTopic/*/* then topic name is myTopic
+        if (topicName.contains("*")) {
+            topicName = topicName.substring(0, topicName.indexOf("*"));
```

Comment:
Can use it, But intellij suggests to use contains. Should I change it to indexOf ?